client/express-checkout/blocks/components/express-checkout-component.js

```diff
@@ -45,12 +73,34 @@ const ExpressCheckoutComponent = ( {
 	const onShippingRateChange = ( event ) =>
 		shippingRateChangeHandler( api, event, elements );
 
+	const onElementsReady = ( event ) => {
+		const paymentMethodContainer = document.getElementById(
+			`express-payment-method-${ PAYMENT_METHOD_NAME_EXPRESS_CHECKOUT_ELEMENT }_${ expressPaymentMethod }`
+		);
+
+		const availablePaymentMethods = event.availablePaymentMethods || {};
+
+		if ( paymentMethodContainer ) {
+			paymentMethodContainer.style.display = availablePaymentMethods[
+				expressPaymentMethod
+			]
+				? null
+				: 'none';
+		}
```

Comment:
Nit: Maybe it's best to only set the display style when we want to hide the element? The rationale here would be to just affect the necessary elements and allow for customizations, but since this would be a far edge case feel free to ignore.

```js
if (
	paymentMethodContainer &&
	! availablePaymentMethods[ expressPaymentMethod ]
) {
	paymentMethodContainer.style.display = 'none';
}
```

Reply:
That makes sense, done in https://github.com/Automattic/woocommerce-payments/pull/9047/commits/806e790cb90d09244ed9dfaf2e5a56eced8d3192